diff --git a/src/main/scala/gemmini/Configs.scala b/src/main/scala/gemmini/Configs.scala
--- a/src/main/scala/gemmini/Configs.scala
+++ b/src/main/scala/gemmini/Configs.scala
@@ -54,7 +54,7 @@
 
     // Reservation station entries
     reservation_station_entries_ld = 8,
-    reservation_station_entries_st = 16,
+    reservation_station_entries_st = 32,
     reservation_station_entries_ex = 16,
 
     // Ld/Ex/St instruction queue lengths
